feature-libs/product-configurator/rulebased/components/attribute/price-change/configurator-attribute-price-change.service.ts

```diff
@@ -0,0 +1,118 @@
+import { ObjectComparisonUtils } from '@spartacus/core';
+import { ConfiguratorRouterExtractorService } from '@spartacus/product-configurator/common';
+import { EMPTY, filter, Observable, of, switchMap, tap } from 'rxjs';
+import { ConfiguratorCommonsService } from '../../../core/facade/configurator-commons.service';
+import { Configurator } from '../../../core/model/configurator.model';
+
+@Injectable()
+export class ConfiguratorAttributePriceChangeService {
+  protected configuratorRouterExtractorService = inject(
+    ConfiguratorRouterExtractorService
+  );
+  protected configuratorCommonsService = inject(ConfiguratorCommonsService);
+  protected isInitialRendering: boolean = true;
+  protected lastAttributeSupplement:
+    | Configurator.AttributeSupplement
+    | undefined;
+  protected valuePrices: { [key: string]: Configurator.PriceDetails } = {};
+
+  /**
+   * Returns an observable that shall be used by all components supporting delta rendering mode.
+   * It will monitor the price supplements of configuration observable and emit true if price supplements
+   * matching the given attribute key have changed.
+   * Additionally it returns always true for the first emission of the underlying configuration observable.
+   * This ensures that an enclosing UI component will initially render, even if the async pricing request ist still running,
+   * so that the UI is not blocked. Afterwards a rerender shall only occur if prices change.
+   * This all assumes that the enclosing UI component itself gets recreated or rerendered (triggered elsewhere) whenever the attribute itself changes content wise.
+   *
+   * @param attributeKey key of the attribute for which the prices should be checked for changes
+   * @returns observable that emits 'true' each time a price changes and hence there is the need to rerender the enclosing component
+   */
+  getPriceChangedEvents(attributeKey: string | undefined): Observable<boolean> {
+    return this.configuratorRouterExtractorService.extractRouterData().pipe(
+      switchMap((routerData) => {
+        return this.configuratorCommonsService
+          .getConfiguration(routerData.owner)
+          .pipe(
+            // Initially render attribute without prices, so UI is not blocked, otherwise only re-ender if prices changed.
+            // Changes of attribute itself are already handled in the attribute composition directive.
+            filter(
+              (config) => this.isInitialRendering || !!config.priceSupplements
+            ),
+            switchMap((config) => {
+              if (!config.priceSupplements) {
+                return of(true);
+              }
+              const pricesChanged = this.checkForValuePriceChanges(
+                config,
+                attributeKey
+              );
+              return pricesChanged ? of(true) : EMPTY;
+            }),
+            tap(() => (this.isInitialRendering = false))
+          );
+      })
```

Comment:
It's not strictly necessary now, but to be future bullet-proof, let's add a shareReplay at the end of the steam (because in this steam we perform "heavy"? deep-equal object comparisons that we don't want to accidentally run more than once, if there are more than 1 subscriber.

``suggestion
      }),
      shareReplay({ bufferSize: 1, refCount: true })
```

Otherwise, if the returned observable `priceChangedEvent$` would be subscribed to in 2 different places (e.g. in 2 different async pipes), then the whole logic of this stream (including switchMap with deep equality check computations inside) will be executed unnecesarily 2 times per each emission from `configuratorCommonsService.getConfiguration`

And I've just noticed that in some components we have 2 async pipes for the same stream `priceChangedEvent$`, see e.g. `configurator-attribute-multi-selection-image.component`:
https://github.com/SAP/spartacus/blob/17ad050c8e339bca93e9d57b146b2e675fbadbd0/feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-image/configurator-attribute-multi-selection-image.component.html#L5
and